lib/rules/no-shadow.js

```diff
@@ -284,6 +484,30 @@ module.exports = {
 			const inner = getNameRange(variable);
 			const outer = getNameRange(scopeVar);
 
+			if (!inner || !outer || inner[1] >= outer[0]) {
+				return false;
+			}
+
+			if (!outerDef) {
+				return true;
+			}
```

Comment:
[Refactor] I believe we're guaranteed `inner` and `outerDef` will always exist:

```suggestion
			if (!outer || inner[1] >= outer[0]) {
				return false;
			}
```